python/src/opendp/extras/polars/__init__.py

```diff
@@ -996,40 +1089,136 @@ class Margin:
     Instances of this class are used by :py:func:`opendp.context.Context.compositor`.
     """
 
-    by: Sequence | None = None
+    by: Sequence = field(default_factory=list)
     """Polars expressions describing the grouping columns."""
 
-    public_info: Literal["keys"] | Literal["lengths"] | None = None
-    """Identifies properties of grouped data that are considered public information.
+    max_length: int | None = None
+    """An upper bound on the number of records in any one group.
+
+    The library may request you set a max dataset or partition length (for instance, for float sums).
+    If you don't know how many records are in the data, you can specify a very loose upper bound,
+    for example, the size of the total population you are sampling from.
+
+    This is used to resolve issues raised in the paper
+    `Widespread Underestimation of Sensitivity in Differentially Private Libraries and How to Fix It <https://arxiv.org/pdf/2207.10635.pdf>`_.
```

Comment:
"The library may request" is a little indirect.

```suggestion
    """Some operations (for instance, for float sums) will error if `max_length` is not provided.
    This is used to resolve issues raised in the paper
    `Widespread Underestimation of Sensitivity in Differentially Private Libraries and How to Fix It <https://arxiv.org/pdf/2207.10635.pdf>`_.
    
        If you don't know how many records are in the data, you can specify a very loose upper bound,
    for example, the size of the total population you are sampling from.
```